src/references.rs

```diff
@@ -299,114 +309,150 @@ impl SchemaPointer {
+                            // value.
+                            return Some(Cow::Owned(result.into_owned()));
                         }
-                        Some(SingleOrVec::Single(_)) => {
-                            eprintln!(
-                                "Warning: trying to access `items` at index {index} in a \
+                    }
+                    SchemaPointerElt::Properties(prop) => current
+                        .object()?
+                        .object
+                        .as_ref()?
+                        .properties
+                        .get(prop)
+                        .map(CurrentSchema::Schema),
+                    SchemaPointerElt::AdditionalProperties => current
+                        .object()?
+                        .object
+                        .as_ref()?
+                        .additional_properties
+                        .as_ref()
+                        .map(|s| CurrentSchema::Schema(s.as_ref())),
+                    SchemaPointerElt::ItemsIndexed(index) => {
+                        match current.object()?.array.as_ref()?.items.as_ref() {
+                            Some(SingleOrVec::Vec(vec)) => {
+                                warn_if_out_of_bounds(vec, *index, "items");
+                                vec.get(*index).map(CurrentSchema::Schema)
+                            }
+                            Some(SingleOrVec::Single(_)) => {
+                                eprintln!(
+                                    "Warning: trying to access `items` at index {index} in a \
                                 reference, but `items` is a single schema in the current \
                                 document, not an array"
-                            );
-                            None
+                                );
+                                None
+                            }
+                            _ => None,
                         }
-                        _ => None,
                     }
-                }
-                SchemaPointerElt::ItemsSingle => match &current.object()?.array.as_ref()?.items {
-                    Some(SingleOrVec::Single(sub)) => Some(CurrentSchema::Schema(sub.as_ref())),
-                    Some(SingleOrVec::Vec(_)) => {
-                        eprintln!(
+                    SchemaPointerElt::ItemsSingle => match &current.object()?.array.as_ref()?.items
+                    {
+                        Some(SingleOrVec::Single(sub)) => Some(CurrentSchema::Schema(sub.as_ref())),
+                        Some(SingleOrVec::Vec(_)) => {
+                            eprintln!(
                             "Warning: trying to access `items` in a reference without an index, \
                             but `items` is an array of schemas in the current document, not \
                             a single schema"
                         );
-                        None
+                            None
```

Comment:
This formatting seems off. Not sure what happened.

Reply:
Yeah, it seems rustfmt doesn't decide the indentation of the arguments and parentheses of `eprintln!`, so if I do something stupid, it doesn't stop me. So the `None` is actually indented correctly, but the format string argument of `eprintln` and the closing `);` should be indented one level to the right. I'll do that manually.